tests/blockchain/eth/entities/agents/test_user_escrow_agent.py

```diff
@@ -0,0 +1,143 @@
+    assert _direct_agent.contract.address == _agent.contract.address
+    assert _agent.principal_contract.address == escrow_deployer.principal_contract.address
+    assert _agent.principal_contract.abi == escrow_deployer.principal_contract.abi
+    assert _direct_agent.contract.abi == escrow_deployer.principal_contract.abi
+    assert _direct_agent.contract.address == escrow_deployer.principal_contract.address
+
+    yield _agent
+    testerchain.interface.registry.clear()
+
+
+def test_user_escrow_agent_represents_beneficiary(agent, three_agents):
+    token_agent, miner_agent, policy_agent = three_agents
+
+    # Name
+    assert agent.registry_contract_name == UserEscrowAgent.registry_contract_name
+
+    # Not Equal to MinerAgent
+    assert agent != miner_agent, "UserEscrow Agent is connected to the MinerEscrow's contract"
+    assert agent.contract_address != miner_agent.contract_address, "UserEscrow and MinerEscrow agents represent the same contract"
+
+    # Proxy Target Accuracy
+    assert agent.principal_contract.address == agent.proxy_contract.address
+    assert agent.principal_contract.abi != agent.proxy_contract.abi
+
+    assert agent.principal_contract.address == agent.contract.address
+    assert agent.principal_contract.abi == agent.contract.abi
+
+
+def test_read_beneficiary(testerchain, agent):
+    deployer_address, beneficiary_address, *everybody_else = testerchain.interface.w3.eth.accounts
+    benficiary = agent.beneficiary
+    assert benficiary == beneficiary_address
+    assert is_checksum_address(benficiary)
+
+
+def test_read_allocation(agent, three_agents):
+    token_agent, miner_agent, policy_agent = three_agents
+    balance = token_agent.get_balance(address=agent.principal_contract.address)
+    assert balance == TEST_ALLOCATION
+    allocation = agent.allocation
+    assert allocation > 0
+    assert allocation == TEST_ALLOCATION
+
+
+def test_read_locked_tokens(agent, three_agents):
+    token_agent, miner_agent, policy_agent = three_agents
+    balance = token_agent.get_balance(address=agent.principal_contract.address)
+    assert balance == TEST_ALLOCATION
+    assert agent.principal_contract.functions.getLockedTokens().call() == TEST_ALLOCATION
+    assert agent.locked_tokens == TEST_ALLOCATION
+
+
+@pytest.mark.usesfixtures("three_agents")
+def test_read_timestamp(agent):
+    timestamp = agent.end_timestamp
+    end_locktime = maya.MayaDT(timestamp)
+    assert end_locktime.slang_time()
+    now = maya.now()
+    assert now < end_locktime, '{} is not in the future!'.format(end_locktime.slang_date())
+
+
+@pytest.mark.usesfixtures("three_agents")
+def test_deposit_as_miner(testerchain, agent, three_agents):
+    old_locked_value = agent.principal_contract.functions.lockedValue().call()
+    txhash = agent.deposit_as_miner(value=MIN_ALLOWED_LOCKED, periods=MIN_LOCKED_PERIODS)
+    assert txhash
+    new_locked_value = agent.principal_contract.functions.lockedValue().call()
```

Comment:
`lockedValue` is unchangeable, `deposit_as_miner` changes stake in `MinersEscrow`